src/theory/splitter.h

```diff
@@ -0,0 +1,138 @@
+/******************************************************************************
+ * Top contributors (to current version):
+ *   Amalee Wilson, Andrew Wu
+ *
+ * This file is part of the cvc5 project.
+ *
+ * Copyright (c) 2009-2021 by the authors listed in the file AUTHORS
+ * in the top-level source directory and their institutional affiliations.
+ * All rights reserved.  See the file COPYING in the top-level source
+ * directory for licensing information.
+ * ****************************************************************************
+ *
+ */
+
+#include "cvc5_private.h"
+
+#ifndef CVC5__THEORY__SPLITTER_H
+#define CVC5__THEORY__SPLITTER_H
+
+#include <math.h>
+
+#include <fstream>
+#include <list>
+#include <sstream>
+
+#include "options/smt_options.h"
+#include "proof/trust_node.h"
+#include "theory/valuation.h"
+
+namespace cvc5 {
+
+class TheoryEngine;
+
+namespace prop {
+class PropEngine;
+}
+
+namespace theory {
+
+class Splitter
+{
+ public:
+  Splitter(TheoryEngine* theoryEngine, prop::PropEngine* propEngine);
+
+  /**
+   * Make partitions for parallel solving.
+   */
+  TrustNode makePartitions();
+
+ private:
+  /**
+   * Print the cube to the specified output.
+   */
+  void emitCube(Node toEmit);
+
+  /**
+   * Block a path in the search by sending the not of toBlock as a lemma to the
+   * SAT solver.
+   */
+  TrustNode blockPath(Node toBlock);
+
+  /**
+   * Close the output file if it is specified.
+   */
+  void closeFile();
+
+  /**
+   * Stop partitioning and return unsat.
+   */
+  TrustNode stopPartitioning();
+
+  /**
+   * Get the list of decisions from the SAT solver
+   */
+  void collectDecisionLiterals(std::vector<TNode>& literals);
+
+  /**
+   * Current propEngine.
+   */
+  prop::PropEngine* d_propEngine;
+
+  /**
+   * Valuation of the theory engine.
+   */
+  std::unique_ptr<Valuation> d_valuation;
+
+  /**
+   * The number of partitions requested through the compute-partitions option.
+   */
+  const unsigned d_numPartitions;
```

Comment:
[We prefer to use types with explicit bit-widths.](https://github.com/cvc5/cvc5/wiki/Code-Style-Guidelines#integer-types) (Also applies to other data members of type `unsigned`)